gnmi/README.md

```diff
@@ -1,22 +1,59 @@
-# Fake gNMI Service (central datastore)
+# Reference gNMI Service (and internal central datastore)
 
-`gnmit.GNMIServer` serves as a central datastore to which subscribing
-"tasks/agents" can be attached.
+`gnmi.GNMIServer` serves as the reference implementation of gNMI for lemming and the central datastore for internal tasks/agents.
 
-There are two ways to subscribe to the central datastore:
+## External Clients
 
-## 1 Subscribing using a `gnmit.Task` interface
+External clients use gNMI to configure and read state of the reference device.
+## Internal Reconcilers
```

Comment:
Internal clients maybe? I guess in general a client doesn't always have to be reconciling between config and internal state. It could be an event handler where whenever the state of the hostname changes to an animal name, then send an HTTP POST request to donate to the WWF, so in that case it would not be reconciling anything, but detecting event triggers. And then later you can introduce "The Reconciler Interface" as a common type of such internal clients.

Are we saying that reconcilers are the only way we interact with the datastore? Is there a way where a user can just use a ygnmi.Client to interact with the datastore without having to understand the reconciler, and the reconciler is just a shorthand? If it's the former, then can this be stated here with a reason why this is?

Reply:
i think everything should implement the reconciler API